File: 0402-remove-k-digits/0402-remove-k-digits.cpp
class Solution {
public:
    string removeKdigits(string num, int k) {
    int n = num.size();
    stack<int>st;
    for(int i=0;i<n;i++){
    while(!st.empty()&&k>0&&st.top()>num[i]){
        st.pop(); 
        k--;
        }
    st.push(num[i]);
    }

    if(k!=0){
        while(k>0){
            st.pop();
            k--;
        }
    }
    string ans;
    while(!st.empty()){
        ans+=st.top();
        st.pop();
    }
    reverse(ans.begin(),ans.end());
    int i=0;
    while(i<n && ans[i]=='0'){
        i++;
    }
    ans=ans.substr(i);
    if(ans.empty()) return "0";
    return ans;
    }
};

File: 0402-remove-k-digits/0402-remove-k-digits.py
class Solution(object):
    def removeKdigits(self, num, k):
        st=[]
        n=len(num)
        for i in range (n):
            while st and k and st[-1]>num[i]:
                st.pop()
                k-=1
            st.append(num[i])
        while k :
            st.pop()
            k-=1
        ans = ''.join(st)
        ans = ans.lstrip('0')
        return ans or "0"